monitored-httpclient/src/test/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/SingleIpHttpClientTest.java

```diff
@@ -102,4 +104,40 @@ void shouldFailOnMalformedUrl() {
         final IllegalArgumentException illegalStateException = assertThrows(IllegalArgumentException.class, () -> new SingleIpHttpClient(httpClient, InetAddress.getLocalHost(), new ServerConfiguration("com.github.nhenneaux.resilienthttpclient.monitoredclientpool.SingleIpHttpClientTest.shouldCreateClientWithoutRefresh", -234, "&dfsfsd", 1, 1, -1)));
         assertEquals("Cannot build health URI from ServerConfiguration{hostname='com.github.nhenneaux.resilienthttpclient.monitoredclientpool.SingleIpHttpClientTest.shouldCreateClientWithoutRefresh', port=-234, healthPath='&dfsfsd', connectionHealthCheckPeriodInSeconds=1, dnsLookupRefreshPeriodInSeconds=1, readTimeoutInMilliseconds=-1}", illegalStateException.getMessage());
     }
+
+    @Test
+    void shouldCallCheckHealthStatusIfHealthyIsFalse() {
+        // Given
+        final String hostname = "cloudflare.com";
+        final HttpClient httpClient = mock(HttpClient.class);
+        ArgumentCaptor<HttpRequest> captor = ArgumentCaptor.forClass(HttpRequest.class);
```

Comment:
I am not sure, this `captor` is used somehow? Or the following mock can be replace with just 
`when(httpClient.sendAsync(any(HttpRequest.class), any(STRING_BODY_HANDLER_CLASS)))...`?

Reply:
Not used at all.